pkg/registry/server/event/controller.go

```diff
@@ -0,0 +1,58 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package event
+
+import (
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/reference"
+
+	modeljobscheme "github.com/caicloud/temp-model-registry/pkg/clientset/clientset/versioned/scheme"
+	"github.com/caicloud/temp-model-registry/pkg/registry/client"
+	"github.com/caicloud/temp-model-registry/pkg/registry/errors"
+	"github.com/caicloud/temp-model-registry/pkg/registry/server/basecontroller"
+)
+
+// eventController for event APIs impl
+type eventController struct {
+	basecontroller.BaseController
+}
+
+func (m *eventController) GetModelJobEvents() {
+	namespace := m.Ctx.Input.Param(":namespace")
+	modeljobID := m.Ctx.Input.Param(":modeljob_id")
+
+	modeljob, err := client.KubeModelJobClient.ModeljobsV1alpha1().ModelJobs(namespace).Get(modeljobID, metav1.GetOptions{})
+	if err != nil {
+		m.RendorError(errors.GeneralCode, err.Error())
+		return
+	}
+
+	var events *corev1.EventList
+	if ref, err := reference.GetReference(scheme.Scheme, modeljob); err != nil {
+		log.Errorf("get modeljob reference err: %v", err)
+	} else {
+		events, err = client.KubeMainClient.CoreV1().Events(namespace).Search(modeljobscheme.Scheme, ref)
```

Comment:
Can we use the informer instead of client to get all resources? Maybe we can open a separate issue to keep track of it. Using the client directly will have performance issues.

Reply:
Ref https://github.com/caicloud/temp-model-registry/issues/29